src/transformers/trainer_tf.py

```diff
@@ -0,0 +1,418 @@
+"""Tensorflow trainer class."""
+
+import logging
+import math
+import os
+from typing import Callable, Dict, Optional
+
+import numpy as np
+import tensorflow as tf
+
+from .modeling_tf_utils import TFPreTrainedModel, shape_list
+from .optimization_tf import GradientAccumulator, create_optimizer
+from .trainer_utils import PREFIX_CHECKPOINT_DIR, EvalPrediction, PredictionOutput, TrainOutput
+from .training_tf_args import TFTrainingArguments
+
+
+logger = logging.getLogger(__name__)
+
+
+class TFDataset:
+    TrainOutput,
+    """
+    Fake superclass to partially imitate the PT Dataset class.
+    """
+
+    def get_dataset(self):
+        return self.dataset
+
+
+class TFTrainer:
+    model: TFPreTrainedModel
+    args: TFTrainingArguments
+    # something similar to a PT Dataset.
+    # This is just temporary before to have
+    # a framework-agnostic approach for datasets.
+    train_dataset: Optional[TFDataset]
+    eval_dataset: Optional[TFDataset]
```

Comment:
Shouldn't we pass `tf.data.Dataset` directly?

And then we don't need the class above?

Reply:
This is temporary, waiting the new lib. I did that mostly to be able to keep the size of a dataset, as it is quite painfull to compute this in TF.

Reply:
So, we will probably launch TFTrainer before the "library-in-stealth", so it should probably be a tf.data.Dataset for now. Will be easier to understand, in my opinion. And we won't have to change anything when the new library is out.

If it's really costly to compute the dataset size, can we maybe just store the size as an attribute on the object in the scripts, and add an error message to the init of the TFTrainer if it's not set?

Reply:
I advocate for having `tf.data.Dataset` here as well!

Reply:
You are right, I will find a hack that will do the job^^